### ISSUE TITLE
NullPointerException from `absolutePath()`

### ISSUE THREAD

**Issue by 3flex** [detekt/detekt-compiler-plugin#15]
```
e: Analyzing 'DbModule.kt' led to an exception.
The original exception message was: root.absolutePath().parent must not be null
Running detekt '1.12.0' on Java '14.0.2+12-46' on OS 'Windows 10'.
If the exception message does not help, please feel free to create an issue on our GitHub page.
e: 

e: io.gitlab.arturbosch.detekt.rules.naming.InvalidPackageDeclaration.postVisit(InvalidPackageDeclaration.kt:45)
```

The problematic line:
https://github.com/detekt/detekt/blob/ab76be9f91aaaab7c0ff7f629d6c55daf03ab08f/detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/InvalidPackageDeclaration.kt#L45
```kotlin
val normalizedFilePath = root.absolutePath().parent.toNormalizedForm()
```

`absolutePath()` extension returns only the filename when run through the compiler plugin, so `absolutePath().parent` is `null` causing the crash.

**Comment by arturbosch** [comment]
Do you have a reproducer?
The compiler returns here a KtFile.name with just `test.kt` right?

**Comment by 3flex** [comment]
Yes, that's right. The issue is in detekt-core, I'm working on a PR.

**Comment by 3flex** [comment]
I've done a bit of experimenting. What I can see is:
1. Detekt & KtLint create in-memory copies of files to be analysed. The in-memory copies are writable, which is required to support KtLint's autocorrect which modifies AST nodes within the writable in-memory KtFile.
2. The Kotlin compiler creates KtFiles which are not writable. When passed to KtLint to autocorrect an exception is thrown.
3. Detekt doesn't respect the semantics of `PsiFile`. It stores an absolute file path where just a filename is expected. `KtFile.getName()` should return a filename but in detekt this will return an absolute file path. Files created by the Kotlin compiler return just the filename - the absolute file path is still available but via a different method. This causes the issue I raised.

I think detekt-core needs to change so it uses the compiler's method of generating KtFiles, then only create a writable copy of it when needed to support autocorrect.

Hopefully that makes sense. I can start a WIP PR so detekt generates the KtFiles the same way as the compiler, but I'm not so familiar with detekt's internals to make the other changes required.
